src/osd/PG.cc

```diff
@@ -5531,6 +5532,7 @@ void PG::handle_advance_map(
   recovery_state.handle_event(evt, rctx);
   if (pool.info.last_change == osdmap_ref->get_epoch())
     on_pool_change();
+  update_store_with_options();
```

Comment:
I checked if collection options update happens on OSD start and it looks like this takes place without any changes to load_pgs. The changes above seems to be enough. Haven't investigated what triggered this though. Do you still think we need load_pgs modification?